### DISCUSSION TITLE
Adding our customary topology attribute to atom object

### DISCUSSION THREAD

**Discussion by ullaharslan** [MDAnalysis/mdanalysis#4607] [Q&A] [answered]
Hello Everyone i hope you are doing great. I using LAMMPS for the analysis of my system, I have performed voronoi analysis and saved the dump with voronoi attribute with "atom_style='id type x y z c_voro[1] c_voro[2]',"
is it possible to add voro[1] in data type and then access the atom group with specified attribute ? Here is my code
```python
import MDAnalysis as mda
u=mda.Universe('data_all_1800_1ns','dump_all_1800_1ns', topology_format="data", format="lammpsdump",atom_style='id type x y z c_voro[1] c_voro[2]',timeunit='ps',dt=0.0001)
grp1=u.select_atoms('c_voro[1] 0 ') # i want to group all atoms with topology attribute c_voro =0
```
any comment suggestoin in this regard will be highly appreciated

**Comment by RMeli** [comment, marked as answer]
Hello @ullaharslan. From the [`LAMMPSParser` documentation](https://docs.mdanalysis.org/1.0.0/documentation_pages/topology/LAMMPSParser.html):

> This should be a space separated
string indicating the position of the `id`, `type`, `resid`, `charge`,
`x`, `y` and `z` fields.  The `resid` and `charge` fields are optional
and **any other specified field will be ignored**.

Have you tried using the optional `resid` and `charge` attribute instead? If you set `id type x y z charge resid`, then `charge` should correspond to your data `c_voro[1]` while `resid` would correspond to your data `c_voro[2]` and your selection would the become `grp1=u.select_atoms('charge 0 ')` instead of `grp1=u.select_atoms('c_voro[1] 0 ')`.

**Reply by RMeli** [reply]
@ullaharslan thanks for the thumbs up. =) If this solves your issue, please formally accept the answer (so that we can keep track of what is solved and what isn't).

**Reply by orbeckst** [reply]
@ullaharslan are you able to *read* the universe? The closed #3504 seems to say that this should work? If so, the data will be in `ts.data["c_voro[1]"]` and you can write code to find the atoms – maybe it's as simple as
```python
ag = u.atoms[ts.data["c_voro[1]"] == 0]]
```
There may also be a way to create a new selectable attribute but that may be more involved for a dynamic property.

**Reply by ullaharslan** [reply]
i am terribly sorry for delay because I was not feeling well. Today I will implement the suggested methodology(That apprantely seems fantastically exact) and will update accordingly. Thank you so much for generous reply. I hope after my checks I will mark it in today date. have a wonderful day.

**Reply by ullaharslan** [reply]
sorry for late reply guys.
Dear @orbeckst off-course I am able to read the universe, but referred suggestion didn't worked actually. atom group command don't recognise the new voronoi attribute. I  believe technically adding an attribute should be possible but not in a simple way.. so we need to do some re work from crash. Suggestion from  @**RMeli tackle the problem in a smart way**.. and I highly appreciate this... so we are using the basic attribute format for our desired ag data.
